std/math/bits/conversion_binary.go

```diff
@@ -69,6 +69,24 @@ func toBinary(api frontend.API, v frontend.Variable, opts ...BaseConversionOptio
 		for i := 0; i < len(bits); i++ {
 			bits[i] = c.Bit(i)
 		}
+
```

Comment:
Why do you need to treat constants differently? Can't we just remove that `if c, ok := api.Compiler().ConstantValue(v); ok`?

Reply:
Either this or imo we can also do `if c.Bitlen() > cfg.NbDigits {panic}`?

Reply:
Oh, Thank you so much for saying that! 😄

I think if we handle constants properly in the lower level functions, like `api.Mul`, `api.Add` and `api.AssertIsEqual`, or if we use some type of optimization pass that optimizes constants in constraints, then we should not need to handle constants in the higher level functions. 

Handling constants in every function can make code very ugly and many higher level functions with high probability will never be called with a constant. (`ToBinary` itself could be an example for that)

Reply:
the part that "worries" me is mostly the hint calls. Calling a hint with `big.Int` is costly in the sense that the inputs to the hint will be serialized with the constraint system, and resolved at solving time. Moreover, the constants will be stored as coefficients in the list of coefficients. That's a lot of gymnastic for something that could be handled at compile time.

Reply:
Interesting... That's a good point.